### PR TITLE
EZP-25822 Do the search request

### PR DESCRIPTION
**Pull request by StephaneDiot**
jira: https://jira.ez.no/browse/EZP-25822
## Description

The user can now do a search. Currentlly only the 10 first results are shown (pagination will come later).
## screencast

https://youtu.be/LIg1UIapT94

Note that CSS & UI will be improved

### CODE REVIEW

**Comment by StephaneDiot** [comment]
ping @dpobel

File: Resources/public/js/views/services/ezsearch-searchviewservice.js [hunk 1]
```diff
@@ -2,21 +2,53 @@ YUI.add('ezsearch-searchviewservice', function (Y) {
...
         _load: function (callback) {
             this.set('searchString', this.get('request').params.searchString);
-            callback();
+
+            if (this.get('searchString')) {
+                this.fire('locationSearch', {
+                    viewName: 'search-' + this.get('searchString'),
+                    resultAttribute: 'searchResultList',
+                    resultTotalCountAttribute: 'searchResultCount',
+                    loadContent: true,
```
Comment (dpobel) [inline Resources/public/js/views/services/ezsearch-searchviewservice.js]: as far as I can see, we don't need the Content, the ContentInfo available in the Location should be enough

File: Resources/public/js/views/ezsearch-searchlistview.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ezsearch-searchlistview', function (Y) {
+    "use strict";
+  
+    Y.namespace('eZSearch');
+
+    Y.eZSearch.SearchListView = Y.Base.create('ezsearchSearchListView', Y.eZ.TemplateBasedView, [], {
+        
+        initializer: function () {
```
Comment (dpobel) [inline Resources/public/js/views/ezsearch-searchlistview.js]: should be removed

File: Resources/public/templates/ezsearchsearchlistview.hbt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                        <td>{{contentType}}</td>
+                        <td>{{translations}}</td>
+                    </tr>
+                {{/each}}
+            </tbody>
+        </table>
+    {{else}}
+        <p class="ez-searchlist-no-sub-items">This search has no result.</p>
+    {{/if}}
+    <div class="ez-subitemlist-loader-mask"></div>
```
Comment (dpobel) [inline Resources/public/templates/ezsearchsearchlistview.hbt]: not necessary

File: Resources/public/templates/ezsearchsearchlistview.hbt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                        <td>{{name}}</td>
+                        <td>{{lastModificationDate}}</td>
+                        <td>{{contentType}}</td>
+                        <td>{{translations}}</td>
+                    </tr>
+                {{/each}}
+            </tbody>
+        </table>
+    {{else}}
+        <p class="ez-searchlist-no-sub-items">This search has no result.</p>
```
Comment (dpobel) [inline Resources/public/templates/ezsearchsearchlistview.hbt]: `no-sub-items` ?

File: Resources/public/templates/ezsearchsearchlistview.hbt [hunk 5]
```diff
@@ -0,0 +1,24 @@
+    {{#if searchResultItems}}
+        <table class="ez-subitemlist-table pure-table pure-table-bordered">
```
Comment (dpobel) [inline Resources/public/templates/ezsearchsearchlistview.hbt]: `ez-subitemlist-table` ?

File: Resources/public/templates/ezsearchsearchlistview.hbt [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            <thead>
+            <tr>
+            {{#each columns}}
+                <th class="ez-subitem-{{ identifier }}-column">{{ name }}</th>
```
Comment (dpobel) [inline Resources/public/templates/ezsearchsearchlistview.hbt]: class prefix with subitem

File: Resources/public/js/views/ezsearch-searchview.js [hunk 7]
```diff
@@ -20,34 +20,31 @@ YUI.add('ezsearch-searchview', function (Y) {
...
-         * Handles the `submitForm` event by preventing the original form to be
-         * submitted by the browser and by changing searchRequest attribute.
-         *
-         * @method _handleFormSubmit
-         * @protected
-         * @param {EventFacade} e
-         */
+        _renderSearchListView: function (view) {
+            this.get('container').one('.ezsearch-searchlist-content').append(
+                view.render().get('container')
```
Comment (dpobel) [inline Resources/public/js/views/ezsearch-searchview.js]: why passing the view in parameter, you can get it from the attribute

File: Resources/public/js/views/ezsearch-searchview.js [hunk 8]
```diff
@@ -20,34 +20,31 @@ YUI.add('ezsearch-searchview', function (Y) {
             this.get('container').setHTML(
                 this.template({
                     "searchString": this.get("searchString"),
+                    "searchResultList": this.get("searchResultList")
```
Comment (dpobel) [inline Resources/public/js/views/ezsearch-searchview.js]: not needed

File: Resources/public/js/views/ezsearch-searchlistview.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            propertyNames: {
+                value: {
+                    'name': 'Name',
+                    'lastModificationDate': 'Modified',
+                    'contentType': 'Content type',
+                    'translations': 'Translations',
+                }
+            },
+            displayedProperties: {
+                value: ['name', 'lastModificationDate', 'contentType', 'translations'],
```
Comment (dpobel) [inline Resources/public/js/views/ezsearch-searchlistview.js]: this gives the impression that you just have to add something here to get a new column in the page but that's not true, better (for now) hard code the columns in the template

File: Resources/public/js/views/ezsearch-searchlistview.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        _setResultItems: function () {
+            var searchResultItems = [];
+
+            if ( !this.get('searchResultList') ) {
+                return this.get('searchResultList');
+            }
+
+            this.get('searchResultList').forEach(function (struct) {
+                searchResultItems.push({
+                    name: struct.content.get('name'),
```
Comment (dpobel) [inline Resources/public/js/views/ezsearch-searchlistview.js]: you should use the `toJSON` on the models available in the struct.

File: Resources/config/yui.yml [hunk 11]
```diff
@@ -11,12 +11,18 @@ system:
...
+                    path: %platformui_search.public_dir%/js/views/ezsearch-searchlistview.js
                 ezsearchsearchview-ez-template:
                     type: 'template'
                     path: %platformui_search.public_dir%/templates/ezsearchsearchview.hbt
+                ezsearchsearchlistview-ez-template:
+                    type: 'template'
+                    path: %platformui_search.public_dir%/templates/ezsearchsearchlistview.hbt
                 ezsearch-searchviewservice:
-                    requires: ['ez-viewservice']
+                    requires: ['ez-viewservice', 'event-valuechange']
```
Comment (dpobel) [inline Resources/config/yui.yml]: valuechange ?

File: Resources/public/js/views/services/ezsearch-searchviewservice.js [hunk 12]
```diff
@@ -2,21 +2,53 @@ YUI.add('ezsearch-searchviewservice', function (Y) {
     Y.namespace('eZSearch');
 
     Y.eZSearch.SearchViewService = Y.Base.create('ezsearchSearchViewService', Y.eZ.ViewService, [], {
+        _initializer: function() {
```
Comment (dpobel) [inline Resources/public/js/views/services/ezsearch-searchviewservice.js]: should be removed

File: Resources/public/js/views/services/ezsearch-searchviewservice.js [hunk 13]
```diff
@@ -2,21 +2,53 @@ YUI.add('ezsearch-searchviewservice', function (Y) {
...
+                callback();
+            }
         },
 
         _getViewParameters: function () {
+            var that = this;
             return {
-                searchString: this.get('searchString'),
+                searchString: that.get('searchString'),
+                searchResultList: this.get('searchResultList')
```
Comment (dpobel) [inline Resources/public/js/views/services/ezsearch-searchviewservice.js]: missing `searchResultCount`

File: Resources/public/templates/ezsearchsearchlistview.hbt [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+            {{/each}}
+            </tr>
+            </thead>
+            <tbody class="ez-searchlist-content">
+                {{#each searchResultItems}}
+                    <tr>
+                        <td>{{name}}</td>
```
Comment (dpobel) [inline Resources/public/templates/ezsearchsearchlistview.hbt]: missing the link to view this content

**Comment by andrerom** [comment]
screen shoot? :)

File: Resources/config/css.yml [hunk 15]
```diff
@@ -4,4 +4,6 @@ system:
             files:
                 - %platformui_search.css_dir%/views/search.css
                 - %platformui_search.css_dir%/theme/views/search.css
-                - %platformui_search.css_dir%/theme/views/actions/button.css
\ No newline at end of file
+                - %platformui_search.css_dir%/theme/views/actions/button.css
+                - %platformui_search.css_dir%/views/searchlist.css
```
Comment (dpobel) [inline Resources/config/css.yml]: please keep the CSS sorted, so layout CSS first, theme ones after

File: Resources/public/js/views/services/ezsearch-searchviewservice.js [hunk 16]
```diff
@@ -2,21 +2,51 @@ YUI.add('ezsearch-searchviewservice', function (Y) {
...
+                this.onceAfter('searchResultListChange', function () {
+                    callback();
+                })
+            } else {
+                callback();
+            }
         },
 
         _getViewParameters: function () {
+            var that = this;
```
Comment (dpobel) [inline Resources/public/js/views/services/ezsearch-searchviewservice.js]: ?

File: Resources/public/js/views/ezsearch-searchlistview.js [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+            }
+
+            this.get('searchResultList').forEach(function (struct) {
+                searchResultItems.push({
+                    content: struct.content.toJSON(),
+                    contentType: struct.contentType.toJSON(),
+                    location: struct.location.toJSON(),
+                });
+            }, this);
+            this._set('searchResultItems', searchResultItems);
```
Comment (dpobel) [inline Resources/public/js/views/ezsearch-searchlistview.js]: I still don't see the point of this attribute, just change that method to be called `_getResultItems` and to return the data ready for the template, and then in `render`, you can directly use its return value.

**Pull request merged**